async-nats/src/jetstream/consumer/push.rs

```diff
@@ -10,19 +10,68 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use std::time::Duration;
+use super::{AckPolicy, Consumer, DeliverPolicy, FromConsumer, IntoConsumerConfig, ReplayPolicy};
+use crate::{
+    jetstream::{self, Context, Message},
+    Error, StatusCode, Subscriber,
+};
 
+use bytes::Bytes;
 use serde::{Deserialize, Serialize};
+use std::pin::Pin;
+use std::task::{self, Poll};
+use std::time::Duration;
 
-use crate::{jetstream, Error};
+impl Consumer<Config> {
+    pub async fn stream(&self) -> Result<Stream, Error> {
+        let deliver_subject = self.info.config.deliver_subject.clone().unwrap();
+        let subscriber = self.context.client.subscribe(deliver_subject).await?;
 
-use super::{AckPolicy, Consumer, DeliverPolicy, FromConsumer, IntoConsumerConfig, ReplayPolicy};
+        Ok(Stream {
+            context: self.context.clone(),
+            subscriber,
+        })
+    }
+}
 
-impl Consumer<Config> {
-    pub fn push(&self) {
-        println!("push");
+pub struct Stream {
+    context: Context,
+    subscriber: Subscriber,
+}
+
+impl futures::Stream for Stream {
+    type Item = Message;
```

Comment:
`Result<Message, Error>`? that would mean tracking publish attempts in response to flow control etc

Reply:
I think that this can be answered as we discussed at client sync: we're aiming to report on any errors on specific consumers/subscriptions instead of silently passing by them or reporting them on registered callback.

So, `Result<Message, Error>` sounds better.
For sure we need that for Pull, and handling those errors differently (i.e. callback on Consumer) would make pretty inconsistent API for no good reason.